base/card-api.gts

```diff
@@ -1072,15 +1100,17 @@ export async function recompute(card: Card): Promise<void> {
 
   // wait a full micro task before we start - this is simple debounce
   await Promise.resolve();
-  if (recomputePromises.get(card) !== recomputePromise) {
-    return;
+  let current: Promise<void> | undefined;
+  if ((current = recomputePromises.get(card)) !== recomputePromise) {
+    return current;
```

Comment:
this was also important in terms of making sure that recompute promises don't go unwatched. If we decide that our promise is not the current one, then we should return the one that is, otherwise we may resolve too early as the current promise that is not ours may still be in flight

Reply:
the thing that watches the not-current promises should be a logger, if they happen to fail we log them